src/packages/cdk/src/app/api.ts

```diff
@@ -0,0 +1,104 @@
+import * as cdk from 'aws-cdk-lib';
+import * as ec2 from 'aws-cdk-lib/aws-ec2';
+import * as lambda from 'aws-cdk-lib/aws-lambda';
+import { NodejsFunction } from 'aws-cdk-lib/aws-lambda-nodejs';
+import { LogGroup } from 'aws-cdk-lib/aws-logs';
+import { AccessLogFormat, LambdaRestApi, LogGroupLogDestination } from 'aws-cdk-lib/aws-apigateway';
+import { Certificate } from 'aws-cdk-lib/aws-certificatemanager';
+import { Construct } from 'constructs';
+
+import { DatabaseStack } from './database';
+
+import { GraphweaverAppConfig } from './types';
+
+export class ApiStack extends cdk.Stack {
+	public readonly lambda: lambda.Function;
+
+	constructor(
+		scope: Construct,
+		id: string,
+		databaseStack: DatabaseStack,
+		config: GraphweaverAppConfig,
+		props?: cdk.StackProps
+	) {
+		super(scope, id, props);
+
+		if (!databaseStack.dbInstance.secret?.secretFullArn)
+			throw new Error('Missing required secret ARN for database');
+
+		// Create GraphQL Lambda Function
+		this.lambda = new NodejsFunction(this, `${id}-api-function`, {
+			runtime: lambda.Runtime.NODEJS_20_X,
+			handler: 'index.handler',
+			entry: require.resolve(config.api.packageName),
+			bundling: {
+				externalModules: [
+					'sqlite3',
+					'better-sqlite3',
+					'pg-query-stream',
+					'tedious',
+					'mysql',
+					'oracledb',
+					'mariadb',
+					'libsql',
+				],
+			},
+			vpc: config.network.vpc,
+			securityGroups: [config.network.graphqlSecurityGroup],
+			vpcSubnets: {
+				subnetType: ec2.SubnetType.PRIVATE_WITH_EGRESS,
+			},
+			memorySize: config.api.memorySize ?? 1024,
+			architecture: lambda.Architecture.ARM_64,
+			environment: {
+				NODE_EXTRA_CA_CERTS: '/var/runtime/ca-cert.pem',
+				DATABASE_SECRET_ARN: databaseStack.dbInstance.secret.secretFullArn,
+				...config.api.envVars,
+			},
+			timeout: cdk.Duration.seconds(config.api.timeout ?? 10),
+		});
+
+		databaseStack.dbInstance.secret.grantRead(this.lambda);
+
+		const apiLogging = new LogGroup(this, `${id}-api-function-logging`);
```

Comment:
I've noticed this in our other CDK stuff. While we prefer ids that are kebab cased, AWS very much prefers pascal casing here. It probably makes sense to go ahead and just adopt the AWS convention, as they strip dashes out of the result so if we started with `api-function-logging` we get `apifunctionlogging` where `APIFunctionLogging` stays as is.